L1Trigger/L1TMuonBarrel/src/L1MuBMTEtaPatternLut.cc

```diff
@@ -0,0 +1,169 @@
+
+#include "FWCore/ParameterSet/interface/FileInPath.h"
+#include "CondFormats/L1TObjects/interface/L1TriggerLutFile.h"
+
+using namespace std;
+
+// --------------------------------
+//       class L1MuDTEtaPatternLut
+//---------------------------------
+
+//----------------
+// Constructors --
+//----------------
+
+L1MuBMTEtaPatternLut::L1MuBMTEtaPatternLut() {
+
+  //  if ( load() != 0 ) {
+  //    cout << "Can not open files to load eta track finder look-up tables for DTTrackFinder!" << endl;
+  //  }
+
+  //  if ( L1MuDTTFConfig::Debug(6) ) print();
+
+}
+
+
+//--------------
+// Destructor --
+//--------------
+
+L1MuBMTEtaPatternLut::~L1MuBMTEtaPatternLut() {
+
+  m_lut.clear();
+
+}
+
+
+//--------------
+// Operations --
+//--------------
+
+//
+// reset look-up table
+//
+void L1MuBMTEtaPatternLut::reset() {
+
+  m_lut.clear();
+
+}
+
+
+//
+// load pattern look-up table for ETF
+//
+int L1MuBMTEtaPatternLut::load() {
+
+  // get directory name
+  string defaultPath = "L1Trigger/";//L1TriggerConfig/DTTrackFinder/parameters/";
+  string eau_dir = "L1TMuon/data/bmtf_luts/LUTs_Ass/";//L1TriggerData/DTTrackFinder/Eau/
+cout<<"skata"<<endl;
+  // assemble file name
+  edm::FileInPath lut_f = edm::FileInPath(string(defaultPath + eau_dir + "ETFPatternList.lut"));
+  string etf_file = lut_f.fullPath();
+
+  // open file
+  L1TriggerLutFile file(etf_file);
+  if ( file.open() != 0 ) return -1;
+  //  if ( L1MuDTTFConfig::Debug(1) ) cout << "Reading file : " 
+  //                                       << file.getName() << endl; 
+
+  // ignore comment lines 
+  file.ignoreLines(16);
+ 
+  // read patterns
+  while ( file.good() ) {
+
+    int id     = file.readInteger();
+    if ( !file.good() ) break;
+    string pat = file.readString();
+    if ( !file.good() ) break;
+    int qual   = file.readInteger();
+    if ( !file.good() ) break;
+    int eta    = file.readInteger();
+    if ( !file.good() ) break;
+    L1MuDTEtaPattern pattern(id,pat,eta,qual);
+      
+    m_lut[pattern.id()] = pattern;
+
+    if ( !file.good() ) { file.close(); break; }
+    
+  }
+
+  file.close();
+    
+  return 0;
+
+}
+
+
+//
+// print pattern look-up table for ETF
+//
+void L1MuBMTEtaPatternLut::print() const {
+
+  cout << endl;
+  cout << "L1 barrel Track Finder ETA Pattern look-up table :" << endl;
+  cout << "==================================================" << endl;
+  cout << endl;
+
+  cout << "ETF Patterns : " <<  m_lut.size() << endl;
+  cout << "======================" << endl;
+  cout << endl;
+
+  LUT::const_iterator iter = m_lut.begin();
+  while ( iter != m_lut.end() ) {
+    cout << (*iter).second << endl;
+    iter++;
+  }
+
+  cout << endl;
+
+}
+
+
+//
+// get pattern with a given ID
+//
+L1MuDTEtaPattern L1MuBMTEtaPatternLut::getPattern(int id) const {
+
+  LUT::const_iterator it = m_lut.find(id);
+  if ( it == m_lut.end() ) {
+    cerr << "Error: L1MuDTEtaPatternLut: pattern not found : " << id << endl;
```

Comment:
use the message logger here

otherwise, what is the range of valid 'id's? eg, why a map rather than a vector?